src/codec.rs

```diff
@@ -552,6 +552,55 @@ impl<T: Decode> Decode for Option<T> {
 	}
 }
 
+const _: () = {
+	use core::num::{
+		NonZeroI8,
+		NonZeroI16,
+		NonZeroI32,
+		NonZeroI64,
+		NonZeroI128,
+		NonZeroU8,
+		NonZeroU16,
+		NonZeroU32,
+		NonZeroU64,
+		NonZeroU128,
+	};
+	macro_rules! impl_for_non_zero {
+		( $( ($name:ty, $wrapped:ty) ),* $(,)? ) => {
+			$(
+				impl Encode for $name {
+					fn size_hint(&self) -> usize {
+						<$wrapped as Encode>::size_hint(&self.get())
+					}
+
+					fn encode_to<W: Output>(&self, dest: &mut W) {
+						<$wrapped as Encode>::encode_to(&self.get(), dest)
```

Comment:
```suggestion
						self.get().encode_to(dest)
```

Reply:
I like explicit library and macro code because it is far more robust to bit rotting, e.g.:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=51f7bb4b11f26565b4fb420647901d95

This fails as soon as one of the primitive types receives a method `encode_to` or a trait with an `encode_to` method comes into play whereas the explicit version prevents this.

Reply:
applied suggestion in latest commit